x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmSettings.java

```diff
@@ -103,6 +127,94 @@ public static String realmSslPrefix(RealmConfig.RealmIdentifier identifier) {
         }).collect(Collectors.toMap(Tuple::v1, Tuple::v2));
     }
 
+    /**
+     * Returns the domain name that the given realm is assigned to.
+     * Assumes {@code #verifyRealmNameToDomainNameAssociation} successfully verified the configuration.
+     */
+    public static @Nullable String getDomainForRealm(Settings globalSettings, RealmConfig.RealmIdentifier realmIdentifier) {
+        // TODO reserved realm settings need to be pulled into core
+        if (realmIdentifier.equals(new RealmConfig.RealmIdentifier("reserved", "reserved"))) {
```

Comment:
Why don't we apply this check to other synthetic realms?

Reply:
Because `reserved` is a valid value for a realm name, that is not the reserved realm. In this case, the realm can be assigned to a domain but the legit reserved realm still must not be assigned to any realm.